src/d_main.c

```diff
@@ -432,29 +432,24 @@ void D_Display (void)
     }
 
   // wipe update
-  wipe_EndScreen(0, 0, video.unscaledw, SCREENHEIGHT);
+  wipe_EndScreen(0, 0, video.width, video.height);
 
   wipestart = I_GetTime () - 1;
 
   do
     {
-      int nowtime, tics;
-      do
-        {
-          I_Sleep(1);
-          nowtime = I_GetTime();
-          tics = nowtime - wipestart;
-        }
-      while (!tics);
-      wipestart = nowtime;
+      int nowtime = I_GetTime();
+      int tics = nowtime - wipestart;
+
+      fractionaltic = I_GetFracTime();
```

Comment:
Are there any side effects to updating `fractionaltic` more than once per frame (usually done only at the top of `D_Display()`)? Would it make more sense or be safer to update a separate variable local to `f_wipe.c`?

Also, any concerns about demo sync?

Reply:
> Are there any side effects to updating `fractionaltic` more than once per frame (usually done only at the top of `D_Display()`)?

For wipe, we only update it in a local loop, once per frame. Overall, I think it's safe, there's just no reason to update it more often than once per frame.

 > Also, any concerns about demo sync?

As long as we spend the same amount of `gametic`s as before, it should be fine.